docs/src/components/ApiReference.tsx

```diff
@@ -1,9 +1,24 @@
 /* eslint-disable react/no-danger */
 import * as React from 'react';
-import { ComponentAPIReference } from 'docs/types/ComponentAPIReference';
 import { type TocEntry } from '@stefanprobst/rehype-extract-toc';
 import classes from './ApiReference.module.css';
 
+export interface ComponentPropReference {
+  name: string;
+  type: {
+    name: string;
+    description: string | undefined;
+  };
+  defaultValue: string | null;
+  description: string;
+}
+
+export interface ComponentAPIReference {
+  name: string;
+  description: string | undefined;
+  props: ComponentPropReference[];
+}
+
```

Comment:
Not a fan of placing type files separately from the code that uses them (colocation of things that change together, etc.)